test/api/model/saved_snippets_test.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:checks/checks.dart';
+import 'package:http/http.dart' as http;
+import 'package:flutter_test/flutter_test.dart';
+import 'package:zulip/api/route/saved_snippets.dart';
+
+import '../../stdlib_checks.dart';
+import '../fake_api.dart';
+import '../route/route_checks.dart';
+
+void main() {
+  test('smoke', () async {
+    return FakeApiConnection.with_((connection) async {
+      connection.prepare(
+        json: CreateSavedSnippetResult(savedSnippetId: 123).toJson());
+      final result = await createSavedSnippet(connection,
+        title: 'test saved snippet', content: 'content');
+      check(result).savedSnippetId.equals(123);
+      check(connection.takeRequests()).single.isA<http.Request>()
+        .bodyFields.deepEquals({
```

Comment:
As long as we're writing this test, it's only a couple more lines to have it check the other key details of the request too: the method and URL path. So let's include those too.